Source/DafnyCore/Compilers/CSharp/CsharpBackend.cs

```diff
@@ -93,54 +93,40 @@ public class CsharpBackend : ExecutableBackend {
     var outputDir = targetFilename == null ? Directory.GetCurrentDirectory() : Path.GetDirectoryName(Path.GetFullPath(targetFilename));
     var outputPath = Path.Join(outputDir, Path.GetFileNameWithoutExtension(Path.GetFileName(dafnyProgramName)) + ".dll");
     var outputJson = Path.Join(outputDir, Path.GetFileNameWithoutExtension(Path.GetFileName(dafnyProgramName)) + ".runtimeconfig.json");
-    if (inMemory) {
```

Comment:
Might be worth mentioning this change of behavior in a release note (that C# always emits source code now). I can't find any note of it in documentation, and I do like making this more consistent with the other backends, but it could be surprising to some customers that have been using Dafny for a long time.